core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -295,13 +300,13 @@ object LAFuture {
     }
   }
 
-  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit) {
+  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit, wrapper:CommonLoanWrapper) {
     val cur = threadInfo.get()
     scheduler.execute(() => {
       val old = threadInfo.get()
       threadInfo.set(cur)
       try {
-        f()
+        wrapper(f())
```

Comment:
(This is more guess than anything else—I'm not sure how good the JVM is at inlining this kind of nested function invocation in cases where it's a no-op vs how good it is at improving branching in cases of a simple conditional check.)